cpp/src/parquet/metadata.h

```diff
@@ -182,6 +182,28 @@ class PARQUET_EXPORT ColumnChunkMetaData {
   std::unique_ptr<ColumnChunkMetaDataImpl> impl_;
 };
 
+/// \brief DataPageStats is a proxy around stats in format::PageHeader.
+class PARQUET_EXPORT DataPageStats {
+ public:
+  static std::unique_ptr<DataPageStats> Make(const void* page_header);
+
+  ~DataPageStats();
+
+  bool Equals(const DataPageStats& other) const;
+
+  int32_t num_values() const;
+  int32_t num_rows() const;
```

Comment:
how is the consumer supposed to know if this is exact?

Reply:
Sorry, this throws an exception if isn't data table V2, the user has no way of knowing that.

Reply:
I think we should expose is_v2_ as well. I will do that.

Reply:
num_rows returns an optional now.

Reply:
> This is great, and valuable information that might not be obvious to a parquet novice. Can you add this fact to the comments for this addition somewhere?

I added this to the comments in the DataPageStats class. please take a look.